### PR TITLE
Bug 1151970 - Rework RemoteClient et al to allow for syncing

### PR DESCRIPTION
**Pull request by rnewman**
This probably needs a DB version bump.

### CODE REVIEW

**Comment by rnewman** [comment]
Now that I'm implementing client and tabs syncing, I'm going further with this, cutting RemoteClient and the RemoteClientsAndTabs interface up to acknowledge that these things appear at different times and different rates. That'll be a later pull request, perhaps.

File: Account/FxALoginStateMachine.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ class FxALoginStateMachine {
         self.client = client
     }
 
-    func advanceFromState(state: FxAState, now: Int64) -> Deferred<FxAState> {
+    func advanceFromState(state: FxAState, now: UInt64) -> Deferred<FxAState> {
```
Comment (st3fan) [inline Account/FxALoginStateMachine.swift]: Do we need a type alias for a sync timestamp type?

File: Client/Frontend/Home/RemoteTabsPanel.swift [hunk 2]
```diff
@@ -93,7 +93,9 @@ class RemoteTabsPanel: UITableViewController, HomePanel {
             view.frame = CGRect(x: 0, y: 0, width: tableView.frame.width, height: RemoteTabsPanelUX.HeaderHeight)
             view.textLabel.text = client.name
             // TODO: allow localization; convert timestamp to relative timestring.
-            view.detailTextLabel.text = "Last synced: \(String(client.lastModified))"
+            // TODO: note that this is very likely to be wrong; it'll show the last time the other device
+            // uploaded a record, *or another device sent that device a command*.
+            view.detailTextLabel.text = "Last synced: \(String(client.modified))"
```
Comment (st3fan) [inline Client/Frontend/Home/RemoteTabsPanel.swift]: Needs to be an NSLocalizedString()

Comment (rnewman) [inline Client/Frontend/Home/RemoteTabsPanel.swift]: This is fixed in Brian's pull request (which might have already landed?).

File: Account/FxALoginStateMachine.swift (hunk 1 above)
Comment (rnewman) [inline Account/FxALoginStateMachine.swift]: Yeah, I'll `typealias` `Timestamp`.

**Comment by rnewman** [comment]
Obsoleted by #328, on top of Sync 1.2 changes.

**Pull request closed without merging**